pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp

```diff
@@ -135,25 +135,50 @@ class NeighborSampler {
 
     // Case 2: Sample with replacement:
     else if (replace) {
-      for (size_t i = 0; i < count; ++i) {
-        const auto edge_id = generator(row_start, row_end);
-        add(edge_id, global_src_node, local_src_node, dst_mapper,
-            out_global_dst_nodes);
+      if (row_end < (1 << 16)) {
+        int arr[count];
+        generator.fill_with_ints(row_start, row_end, count, &arr[0]);
```

Comment:
I think we should have an overloaded version of `operator()` specialized for mkl. That will help to have a single loop for both cases. Data for mkl case could be kept in `std::vector` on the `RandintEngine` side.

Reply:
Update: let's leave it for future enhancements.